src/storage/src/source/kinesis.rs

```diff
@@ -213,101 +211,126 @@ impl SourceReader for KinesisSourceReader {
                 .block_on(self.update_shard_information())
             {
                 error!("{:#?}", e);
-                return Err(SourceReaderError::other_definite(e));
+                // XXX(petrosagg): We are fabricating a timestamp here. Is the error truly
+                // definite?
```

Comment:
One alternative, for the sorts of errors we're confident it would be fine to retry, is to return an errored `SourceStatusUpdate`. Example of how that looks here: https://github.com/MaterializeInc/materialize/pull/16156

I'd be okay punting that to the future if you prefer though; you are not making things worse and the concern is clearly noted.

Ditto for the other XXXs.